runtime/format.cc

```diff
@@ -7,23 +7,43 @@
 //===----------------------------------------------------------------------===//
 
 #include "format.h"
+#include "io-stmt.h"
 #include "../lib/common/format.h"
 #include "../lib/decimal/decimal.h"
 #include <limits>
 
-namespace Fortran::runtime {
+namespace Fortran::runtime::io {
+
+// Default FormatContext virtual member functions
+void FormatContext::Emit(const char *, std::size_t) {
+  Crash("Cannot emit data from this FORMAT string");
+}
```

Comment:
I am a bit lost as regarding to when these messages should be emitted.
Are the internal error (i.e the runtime engineer should have overridden this), or are they also intended to handle user error ?
In the first case, I would suggest adding "Internal error" or something similar.
In the second case, the `FormatContext` does not have access to an `IoErrorHandler`, so these crashes cannot be prevented if the user used "ERR", right ?

Reply:
`READ(unit,"(`text')") x` would elicit this error; since the format string could be a run-time variable, we can't always validate it until execution.  These errors cannot be caught with `ERR=/IOSTAT=` (although that would be a reasonable extension).